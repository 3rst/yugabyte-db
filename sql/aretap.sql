\unset ECHO
\i test_setup.sql

SELECT plan(90);
--SELECT * FROM no_plan();

-- This will be rolled back. :-)
SET client_min_messages = warning;

CREATE TABLE public.fou(
    id    INT NOT NULL PRIMARY KEY,
    name  TEXT DEFAULT '',
    numb  NUMERIC(10, 2),
    myint NUMERIC(8)
);
CREATE TABLE public.foo(
    id    INT NOT NULL PRIMARY KEY
);
CREATE TYPE public.sometype AS (
    id    INT,
    name  TEXT
);

CREATE VIEW voo AS SELECT * FROM foo;
CREATE VIEW vou AS SELECT * FROM fou;

CREATE SEQUENCE public.someseq;
CREATE SEQUENCE public.sumeseq;

CREATE SCHEMA someschema;
RESET client_min_messages;

/****************************************************************************/
-- Test tables_are().
SELECT * FROM check_test(
    tables_are( 'public', ARRAY['fou', 'foo'], 'whatever' ),
    true,
    'tables_are(schema, tables, desc)',
    'whatever',
    ''
);

SELECT * FROM check_test(
    tables_are( 'public', ARRAY['fou', 'foo'] ),
    true,
    'tables_are(schema, tables)',
    'Schema public should have the correct tables',
    ''
);

SELECT * FROM check_test(
    tables_are( ARRAY['fou', 'foo'] ),
    true,
    'tables_are(tables)',
    'There should be the correct tables',
    ''
);

SELECT * FROM check_test(
    tables_are( ARRAY['fou', 'foo'], 'whatever' ),
    true,
    'tables_are(tables, desc)',
    'whatever',
    ''
);

SELECT * FROM check_test(
    tables_are( 'public', ARRAY['fou', 'foo', 'bar'] ),
    false,
    'tables_are(schema, tables) missing',
    'Schema public should have the correct tables',
    '    Schema public is missing these tables:
        bar'
);

SELECT * FROM check_test(
    tables_are( ARRAY['fou', 'foo', 'bar'] ),
    false,
    'tables_are(tables) missing',
    'There should be the correct tables',
    '    These tables are missing:
        bar'
);

SELECT * FROM check_test(
    tables_are( 'public', ARRAY['fou'] ),
    false,
    'tables_are(schema, tables) extra',
    'Schema public should have the correct tables',
    '    Schema public has these extra tables:
        foo'
);

SELECT * FROM check_test(
    tables_are( ARRAY['fou'] ),
    false,
    'tables_are(tables) extra',
    'There should be the correct tables',
    '    These are extra tables:
        foo'
);

SELECT * FROM check_test(
    tables_are( 'public', ARRAY['bar', 'baz'] ),
    false,
    'tables_are(schema, tables) extra and missing',
    'Schema public should have the correct tables',
    '    Schema public has these extra tables:
        fo[ou]
        fo[ou]
    Schema public is missing these tables:
        ba[rz]
        ba[rz]',
    true
);

SELECT * FROM check_test(
    tables_are( ARRAY['bar', 'baz'] ),
    false,
    'tables_are(tables) extra and missing',
    'There should be the correct tables',
    '    These are extra tables:
        fo[ou]
        fo[ou]
    These tables are missing:
        ba[rz]
        ba[rz]',
    true
);

/****************************************************************************/
-- Test views_are().
SELECT * FROM check_test(
    views_are( 'public', ARRAY['vou', 'voo'], 'whatever' ),
    true,
    'views_are(schema, views, desc)',
    'whatever',
    ''
);

SELECT * FROM check_test(
    views_are( 'public', ARRAY['vou', 'voo'] ),
    true,
    'views_are(schema, views)',
    'Schema public should have the correct views',
    ''
);

SELECT * FROM check_test(
    views_are( ARRAY['vou', 'voo'] ),
    true,
    'views_are(views)',
    'There should be the correct views',
    ''
);

SELECT * FROM check_test(
    views_are( ARRAY['vou', 'voo'], 'whatever' ),
    true,
    'views_are(views, desc)',
    'whatever',
    ''
);

SELECT * FROM check_test(
    views_are( 'public', ARRAY['vou', 'voo', 'bar'] ),
    false,
    'views_are(schema, views) missing',
    'Schema public should have the correct views',
    '    Schema public is missing these views:
        bar'
);

SELECT * FROM check_test(
    views_are( ARRAY['vou', 'voo', 'bar'] ),
    false,
    'views_are(views) missing',
    'There should be the correct views',
    '    These views are missing:
        bar'
);

SELECT * FROM check_test(
    views_are( 'public', ARRAY['vou'] ),
    false,
    'views_are(schema, views) extra',
    'Schema public should have the correct views',
    '    Schema public has these extra views:
        voo'
);

SELECT * FROM check_test(
    views_are( ARRAY['vou'] ),
    false,
    'views_are(views) extra',
    'There should be the correct views',
    '    These are extra views:
        voo'
);

SELECT * FROM check_test(
    views_are( 'public', ARRAY['bar', 'baz'] ),
    false,
    'views_are(schema, views) extra and missing',
    'Schema public should have the correct views',
    '    Schema public has these extra views:
        vo[ou]
        vo[ou]
    Schema public is missing these views:
        ba[rz]
        ba[rz]',
    true
);

SELECT * FROM check_test(
    views_are( ARRAY['bar', 'baz'] ),
    false,
    'views_are(views) extra and missing',
    'There should be the correct views',
    '    These are extra views:
        vo[ou]
        vo[ou]
    These views are missing:
        ba[rz]
        ba[rz]',
    true
);

/****************************************************************************/
-- Test sequences_are().
SELECT * FROM check_test(
    sequences_are( 'public', ARRAY['sumeseq', 'someseq'], 'whatever' ),
    true,
    'sequences_are(schema, sequences, desc)',
    'whatever',
    ''
);

SELECT * FROM check_test(
    sequences_are( 'public', ARRAY['sumeseq', 'someseq'] ),
    true,
    'sequences_are(schema, sequences)',
    'Schema public should have the correct sequences',
    ''
);

SELECT * FROM check_test(
    sequences_are( ARRAY['sumeseq', 'someseq'] ),
    true,
    'sequences_are(sequences)',
    'There should be the correct sequences',
    ''
);

SELECT * FROM check_test(
    sequences_are( ARRAY['sumeseq', 'someseq'], 'whatever' ),
    true,
    'sequences_are(sequences, desc)',
    'whatever',
    ''
);

SELECT * FROM check_test(
    sequences_are( 'public', ARRAY['sumeseq', 'someseq', 'bar'] ),
    false,
    'sequences_are(schema, sequences) missing',
    'Schema public should have the correct sequences',
    '    Schema public is missing these sequences:
        bar'
);

SELECT * FROM check_test(
    sequences_are( ARRAY['sumeseq', 'someseq', 'bar'] ),
    false,
    'sequences_are(sequences) missing',
    'There should be the correct sequences',
    '    These sequences are missing:
        bar'
);

SELECT * FROM check_test(
    sequences_are( 'public', ARRAY['sumeseq'] ),
    false,
    'sequences_are(schema, sequences) extra',
    'Schema public should have the correct sequences',
    '    Schema public has these extra sequences:
        someseq'
);

SELECT * FROM check_test(
    sequences_are( ARRAY['sumeseq'] ),
    false,
    'sequences_are(sequences) extra',
    'There should be the correct sequences',
    '    These are extra sequences:
        someseq'
);

SELECT * FROM check_test(
    sequences_are( 'public', ARRAY['bar', 'baz'] ),
    false,
    'sequences_are(schema, sequences) extra and missing',
    'Schema public should have the correct sequences',
    '    Schema public has these extra sequences:
        s[ou]meseq
        s[ou]meseq
    Schema public is missing these sequences:
        ba[rz]
        ba[rz]',
    true
);

SELECT * FROM check_test(
    sequences_are( ARRAY['bar', 'baz'] ),
    false,
    'sequences_are(sequences) extra and missing',
    'There should be the correct sequences',
    '    These are extra sequences:
        s[ou]meseq
        s[ou]meseq
    These sequences are missing:
        ba[rz]
        ba[rz]',
    true
);

/****************************************************************************/
-- Finish the tests and clean up.
SELECT * FROM finish();
ROLLBACK;
