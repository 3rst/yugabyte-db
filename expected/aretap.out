\unset ECHO
1..250
ok 1 - tablespaces_are(schemas, desc) should pass
ok 2 - tablespaces_are(schemas, desc) should have the proper description
ok 3 - tablespaces_are(schemas, desc) should have the proper diagnostics
ok 4 - tablespaces_are(schemas) should pass
ok 5 - tablespaces_are(schemas) should have the proper description
ok 6 - tablespaces_are(schemas) should have the proper diagnostics
ok 7 - tablespaces_are(schemas, desc) missing should fail
ok 8 - tablespaces_are(schemas, desc) missing should have the proper description
ok 9 - tablespaces_are(schemas, desc) missing should have the proper diagnostics
ok 10 - tablespaces_are(schemas, desc) extra should fail
ok 11 - tablespaces_are(schemas, desc) extra should have the proper description
ok 12 - tablespaces_are(schemas, desc) extra should have the proper diagnostics
ok 13 - tablespaces_are(schemas, desc) extras and missing should fail
ok 14 - tablespaces_are(schemas, desc) extras and missing should have the proper description
ok 15 - tablespaces_are(schemas, desc) extras and missing should have the proper diagnostics
ok 16 - schemas_are(schemas, desc) should pass
ok 17 - schemas_are(schemas, desc) should have the proper description
ok 18 - schemas_are(schemas, desc) should have the proper diagnostics
ok 19 - schemas_are(schemas) should pass
ok 20 - schemas_are(schemas) should have the proper description
ok 21 - schemas_are(schemas) should have the proper diagnostics
ok 22 - schemas_are(schemas, desc) missing should fail
ok 23 - schemas_are(schemas, desc) missing should have the proper description
ok 24 - schemas_are(schemas, desc) missing should have the proper diagnostics
ok 25 - schemas_are(schemas, desc) extras should fail
ok 26 - schemas_are(schemas, desc) extras should have the proper description
ok 27 - schemas_are(schemas, desc) extras should have the proper diagnostics
ok 28 - schemas_are(schemas, desc) missing and extras should fail
ok 29 - schemas_are(schemas, desc) missing and extras should have the proper description
ok 30 - schemas_are(schemas, desc) missing and extras should have the proper diagnostics
ok 31 - tables_are(schema, tables, desc) should pass
ok 32 - tables_are(schema, tables, desc) should have the proper description
ok 33 - tables_are(schema, tables, desc) should have the proper diagnostics
ok 34 - tables_are(schema, tables) should pass
ok 35 - tables_are(schema, tables) should have the proper description
ok 36 - tables_are(schema, tables) should have the proper diagnostics
ok 37 - tables_are(tables) should pass
ok 38 - tables_are(tables) should have the proper description
ok 39 - tables_are(tables) should have the proper diagnostics
ok 40 - tables_are(tables, desc) should pass
ok 41 - tables_are(tables, desc) should have the proper description
ok 42 - tables_are(tables, desc) should have the proper diagnostics
ok 43 - tables_are(schema, tables) missing should fail
ok 44 - tables_are(schema, tables) missing should have the proper description
ok 45 - tables_are(schema, tables) missing should have the proper diagnostics
ok 46 - tables_are(tables) missing should fail
ok 47 - tables_are(tables) missing should have the proper description
ok 48 - tables_are(tables) missing should have the proper diagnostics
ok 49 - tables_are(schema, tables) extra should fail
ok 50 - tables_are(schema, tables) extra should have the proper description
ok 51 - tables_are(schema, tables) extra should have the proper diagnostics
ok 52 - tables_are(tables) extra should fail
ok 53 - tables_are(tables) extra should have the proper description
ok 54 - tables_are(tables) extra should have the proper diagnostics
ok 55 - tables_are(schema, tables) extra and missing should fail
ok 56 - tables_are(schema, tables) extra and missing should have the proper description
ok 57 - tables_are(schema, tables) extra and missing should have the proper diagnostics
ok 58 - tables_are(tables) extra and missing should fail
ok 59 - tables_are(tables) extra and missing should have the proper description
ok 60 - tables_are(tables) extra and missing should have the proper diagnostics
ok 61 - views_are(schema, views, desc) should pass
ok 62 - views_are(schema, views, desc) should have the proper description
ok 63 - views_are(schema, views, desc) should have the proper diagnostics
ok 64 - views_are(schema, views) should pass
ok 65 - views_are(schema, views) should have the proper description
ok 66 - views_are(schema, views) should have the proper diagnostics
ok 67 - views_are(views) should pass
ok 68 - views_are(views) should have the proper description
ok 69 - views_are(views) should have the proper diagnostics
ok 70 - views_are(views, desc) should pass
ok 71 - views_are(views, desc) should have the proper description
ok 72 - views_are(views, desc) should have the proper diagnostics
ok 73 - views_are(schema, views) missing should fail
ok 74 - views_are(schema, views) missing should have the proper description
ok 75 - views_are(schema, views) missing should have the proper diagnostics
ok 76 - views_are(views) missing should fail
ok 77 - views_are(views) missing should have the proper description
ok 78 - views_are(views) missing should have the proper diagnostics
ok 79 - views_are(schema, views) extra should fail
ok 80 - views_are(schema, views) extra should have the proper description
ok 81 - views_are(schema, views) extra should have the proper diagnostics
ok 82 - views_are(views) extra should fail
ok 83 - views_are(views) extra should have the proper description
ok 84 - views_are(views) extra should have the proper diagnostics
ok 85 - views_are(schema, views) extra and missing should fail
ok 86 - views_are(schema, views) extra and missing should have the proper description
ok 87 - views_are(schema, views) extra and missing should have the proper diagnostics
ok 88 - views_are(views) extra and missing should fail
ok 89 - views_are(views) extra and missing should have the proper description
ok 90 - views_are(views) extra and missing should have the proper diagnostics
ok 91 - sequences_are(schema, sequences, desc) should pass
ok 92 - sequences_are(schema, sequences, desc) should have the proper description
ok 93 - sequences_are(schema, sequences, desc) should have the proper diagnostics
ok 94 - sequences_are(schema, sequences) should pass
ok 95 - sequences_are(schema, sequences) should have the proper description
ok 96 - sequences_are(schema, sequences) should have the proper diagnostics
ok 97 - sequences_are(sequences) should pass
ok 98 - sequences_are(sequences) should have the proper description
ok 99 - sequences_are(sequences) should have the proper diagnostics
ok 100 - sequences_are(sequences, desc) should pass
ok 101 - sequences_are(sequences, desc) should have the proper description
ok 102 - sequences_are(sequences, desc) should have the proper diagnostics
ok 103 - sequences_are(schema, sequences) missing should fail
ok 104 - sequences_are(schema, sequences) missing should have the proper description
ok 105 - sequences_are(schema, sequences) missing should have the proper diagnostics
ok 106 - sequences_are(sequences) missing should fail
ok 107 - sequences_are(sequences) missing should have the proper description
ok 108 - sequences_are(sequences) missing should have the proper diagnostics
ok 109 - sequences_are(schema, sequences) extra should fail
ok 110 - sequences_are(schema, sequences) extra should have the proper description
ok 111 - sequences_are(schema, sequences) extra should have the proper diagnostics
ok 112 - sequences_are(sequences) extra should fail
ok 113 - sequences_are(sequences) extra should have the proper description
ok 114 - sequences_are(sequences) extra should have the proper diagnostics
ok 115 - sequences_are(schema, sequences) extra and missing should fail
ok 116 - sequences_are(schema, sequences) extra and missing should have the proper description
ok 117 - sequences_are(schema, sequences) extra and missing should have the proper diagnostics
ok 118 - sequences_are(sequences) extra and missing should fail
ok 119 - sequences_are(sequences) extra and missing should have the proper description
ok 120 - sequences_are(sequences) extra and missing should have the proper diagnostics
ok 121 - functions_are(schema, functions, desc) should pass
ok 122 - functions_are(schema, functions, desc) should have the proper description
ok 123 - functions_are(schema, functions, desc) should have the proper diagnostics
ok 124 - functions_are(schema, functions) should pass
ok 125 - functions_are(schema, functions) should have the proper description
ok 126 - functions_are(schema, functions, desc) + missing should fail
ok 127 - functions_are(schema, functions, desc) + missing should have the proper description
ok 128 - functions_are(schema, functions, desc) + missing should have the proper diagnostics
ok 129 - functions_are(schema, functions, desc) + extra should fail
ok 130 - functions_are(schema, functions, desc) + extra should have the proper description
ok 131 - functions_are(schema, functions, desc) + extra should have the proper diagnostics
ok 132 - functions_are(schema, functions, desc) + extra & missing should fail
ok 133 - functions_are(schema, functions, desc) + extra & missing should have the proper description
ok 134 - functions_are(schema, functions, desc) + extra & missing should have the proper diagnostics
ok 135 - functions_are(functions, desc) should pass
ok 136 - functions_are(functions, desc) should have the proper description
ok 137 - functions_are(functions, desc) should have the proper diagnostics
ok 138 - functions_are(functions) should pass
ok 139 - functions_are(functions) should have the proper description
ok 140 - functions_are(functions) should have the proper diagnostics
ok 141 - functions_are(functions, desc) + missing should fail
ok 142 - functions_are(functions, desc) + missing should have the proper description
ok 143 - functions_are(functions, desc) + missing should have the proper diagnostics
ok 144 - functions_are(functions, desc) + extra should fail
ok 145 - functions_are(functions, desc) + extra should have the proper description
ok 146 - functions_are(functions, desc) + extra should have the proper diagnostics
ok 147 - functions_are(functions, desc) + extra & missing should fail
ok 148 - functions_are(functions, desc) + extra & missing should have the proper description
ok 149 - functions_are(functions, desc) + extra & missing should have the proper diagnostics
ok 150 - indexes_are(schema, table, indexes, desc) should pass
ok 151 - indexes_are(schema, table, indexes, desc) should have the proper description
ok 152 - indexes_are(schema, table, indexes, desc) should have the proper diagnostics
ok 153 - indexes_are(schema, table, indexes) should pass
ok 154 - indexes_are(schema, table, indexes) should have the proper description
ok 155 - indexes_are(schema, table, indexes) should have the proper diagnostics
ok 156 - indexes_are(schema, table, indexes) + extra should fail
ok 157 - indexes_are(schema, table, indexes) + extra should have the proper description
ok 158 - indexes_are(schema, table, indexes) + extra should have the proper diagnostics
ok 159 - indexes_are(schema, table, indexes) + missing should fail
ok 160 - indexes_are(schema, table, indexes) + missing should have the proper description
ok 161 - indexes_are(schema, table, indexes) + missing should have the proper diagnostics
ok 162 - indexes_are(schema, table, indexes) + extra & missing should fail
ok 163 - indexes_are(schema, table, indexes) + extra & missing should have the proper description
ok 164 - indexes_are(schema, table, indexes) + extra & missing should have the proper diagnostics
ok 165 - indexes_are(table, indexes, desc) should pass
ok 166 - indexes_are(table, indexes, desc) should have the proper description
ok 167 - indexes_are(table, indexes, desc) should have the proper diagnostics
ok 168 - indexes_are(table, indexes) should pass
ok 169 - indexes_are(table, indexes) should have the proper description
ok 170 - indexes_are(table, indexes) should have the proper diagnostics
ok 171 - indexes_are(table, indexes) + extra should fail
ok 172 - indexes_are(table, indexes) + extra should have the proper description
ok 173 - indexes_are(table, indexes) + extra should have the proper diagnostics
ok 174 - indexes_are(table, indexes) + missing should fail
ok 175 - indexes_are(table, indexes) + missing should have the proper description
ok 176 - indexes_are(table, indexes) + missing should have the proper diagnostics
ok 177 - indexes_are(table, indexes) + extra & missing should fail
ok 178 - indexes_are(table, indexes) + extra & missing should have the proper description
ok 179 - indexes_are(table, indexes) + extra & missing should have the proper diagnostics
ok 180 - users_are(users, desc) should pass
ok 181 - users_are(users, desc) should have the proper description
ok 182 - users_are(users, desc) should have the proper diagnostics
ok 183 - users_are(users) should pass
ok 184 - users_are(users) should have the proper description
ok 185 - users_are(users) should have the proper diagnostics
ok 186 - users_are(users, desc) missing should fail
ok 187 - users_are(users, desc) missing should have the proper description
ok 188 - users_are(users, desc) missing should have the proper diagnostics
ok 189 - users_are(users, desc) extras should fail
ok 190 - users_are(users, desc) extras should have the proper description
ok 191 - users_are(users, desc) extras should have the proper diagnostics
ok 192 - users_are(users, desc) missing and extras should fail
ok 193 - users_are(users, desc) missing and extras should have the proper description
ok 194 - users_are(users, desc) missing and extras should have the proper diagnostics
ok 195 - groups_are(groups, desc) should pass
ok 196 - groups_are(groups, desc) should have the proper description
ok 197 - groups_are(groups, desc) should have the proper diagnostics
ok 198 - groups_are(groups) should pass
ok 199 - groups_are(groups) should have the proper description
ok 200 - groups_are(groups) should have the proper diagnostics
ok 201 - groups_are(groups, desc) missing should fail
ok 202 - groups_are(groups, desc) missing should have the proper description
ok 203 - groups_are(groups, desc) missing should have the proper diagnostics
ok 204 - groups_are(groups, desc) extras should fail
ok 205 - groups_are(groups, desc) extras should have the proper description
ok 206 - groups_are(groups, desc) extras should have the proper diagnostics
ok 207 - groups_are(groups, desc) missing and extras should fail
ok 208 - groups_are(groups, desc) missing and extras should have the proper description
ok 209 - groups_are(groups, desc) missing and extras should have the proper diagnostics
ok 210 - languages_are(languages, desc) should pass
ok 211 - languages_are(languages, desc) should have the proper description
ok 212 - languages_are(languages, desc) should have the proper diagnostics
ok 213 - languages_are(languages) should pass
ok 214 - languages_are(languages) should have the proper description
ok 215 - languages_are(languages) should have the proper diagnostics
ok 216 - languages_are(languages, desc) missing should fail
ok 217 - languages_are(languages, desc) missing should have the proper description
ok 218 - languages_are(languages, desc) missing should have the proper diagnostics
ok 219 - languages_are(languages, desc) extras should fail
ok 220 - languages_are(languages, desc) extras should have the proper description
ok 221 - languages_are(languages, desc) extras should have the proper diagnostics
ok 222 - languages_are(languages, desc) missing and extras should fail
ok 223 - languages_are(languages, desc) missing and extras should have the proper description
ok 224 - languages_are(languages, desc) missing and extras should have the proper diagnostics
ok 225 - opclasses_are(schema, opclasses, desc) should pass
ok 226 - opclasses_are(schema, opclasses, desc) should have the proper description
ok 227 - opclasses_are(schema, opclasses, desc) should have the proper diagnostics
ok 228 - opclasses_are(schema, opclasses) should pass
ok 229 - opclasses_are(schema, opclasses) should have the proper description
ok 230 - opclasses_are(schema, opclasses, desc) + missing should fail
ok 231 - opclasses_are(schema, opclasses, desc) + missing should have the proper description
ok 232 - opclasses_are(schema, opclasses, desc) + missing should have the proper diagnostics
ok 233 - opclasses_are(schema, opclasses, desc) + extra should fail
ok 234 - opclasses_are(schema, opclasses, desc) + extra should have the proper description
ok 235 - opclasses_are(schema, opclasses, desc) + extra should have the proper diagnostics
ok 236 - opclasses_are(schema, opclasses, desc) + extra & missing should fail
ok 237 - opclasses_are(schema, opclasses, desc) + extra & missing should have the proper description
ok 238 - opclasses_are(schema, opclasses, desc) + extra & missing should have the proper diagnostics
ok 239 - opclasses_are(opclasses) should pass
ok 240 - opclasses_are(opclasses) should have the proper description
ok 241 - opclasses_are(opclasses) should have the proper diagnostics
ok 242 - opclasses_are(opclasses, desc) + missing should fail
ok 243 - opclasses_are(opclasses, desc) + missing should have the proper description
ok 244 - opclasses_are(opclasses, desc) + missing should have the proper diagnostics
ok 245 - opclasses_are(opclasses, desc) + extra should fail
ok 246 - opclasses_are(opclasses, desc) + extra should have the proper description
ok 247 - opclasses_are(opclasses, desc) + extra should have the proper diagnostics
ok 248 - opclasses_are(opclasses, desc) + extra & missing should fail
ok 249 - opclasses_are(opclasses, desc) + extra & missing should have the proper description
ok 250 - opclasses_are(opclasses, desc) + extra & missing should have the proper diagnostics
